examples/linq.rs

```diff
@@ -48,52 +49,27 @@ impl SinkOperator {
     }
 }
 
-struct SinkOperatorState {
-    message_counter: HashMap<Timestamp, usize>,
-    current_timestamp: Timestamp,
-}
-
-impl SinkOperatorState {
-    fn new() -> Self {
-        Self {
-            message_counter: HashMap::new(),
-            current_timestamp: Timestamp::Bottom,
-        }
-    }
-
-    fn increment_message_count(&mut self, timestamp: &Timestamp) {
-        let count = self.message_counter.entry(timestamp.clone()).or_insert(0);
-        *count += 1;
-    }
-
-    fn get_message_count(&self, timestamp: &Timestamp) -> usize {
-        *self.message_counter.get(timestamp).unwrap_or_else(|| &0)
-    }
-}
-
-impl StateT for SinkOperatorState {
-    fn commit(&mut self, timestamp: &Timestamp) {
-        self.current_timestamp = timestamp.clone();
-    }
-
-    fn get_last_committed_timestamp(&self) -> Timestamp {
-        self.current_timestamp.clone()
-    }
-}
-
-impl Sink<SinkOperatorState, usize> for SinkOperator {
-    fn on_data(&mut self, ctx: &mut SinkContext<SinkOperatorState>, data: &usize) {
+impl Sink<TimeVersionedState<usize>, usize> for SinkOperator {
+    fn on_data(&mut self, ctx: &mut SinkContext<TimeVersionedState<usize>>, data: &usize) {
         let timestamp = ctx.get_timestamp().clone();
-        tracing::info!("SinkOperator @ {:?}: Received {}", timestamp, data);
-        ctx.get_state().increment_message_count(&timestamp);
+        tracing::info!(
+            "{} @ {:?}: Received {}",
+            ctx.get_operator_config().get_name(),
+            timestamp,
+            data,
+        );
+
+        // Increment the message count.
+        *ctx.get_current_state().unwrap() += 1;
     }
 
-    fn on_watermark(&mut self, ctx: &mut SinkContext<SinkOperatorState>) {
+    fn on_watermark(&mut self, ctx: &mut SinkContext<TimeVersionedState<usize>>) {
         let timestamp = ctx.get_timestamp().clone();
         tracing::info!(
-            "SinkOperator @ {:?}: Received {} data messages.",
+            "{} @ {:?}: Received {} data messages.",
+            ctx.get_operator_config().get_name(),
             timestamp,
-            ctx.get_state().get_message_count(&timestamp),
+            *ctx.get_current_state().unwrap(),
```

Comment:
```suggestion
            ctx.get_current_state().unwrap(),
```
Dereferencing shouldn't be necessary for displaying the value.